service/history/api/create_workflow_util.go

```diff
@@ -187,6 +187,7 @@ func NewWorkflowVersionCheck(
 		return nil
 	}
 
+	// we are creating a new workflow, so workflow must be running here,
```

Comment:
NOTE for reviewers: I added a comment at all usages of mutableState.GetCurrentVersion() explaining why updating workflow currentVersion is safe.
Those comments will be removed before merging.